application/features/bootstrap/SyncContext.php

```diff
@@ -377,4 +398,17 @@ public function usersAreActiveInTheIdStoreAndAreInactiveInTheIdBroker($number)
         }
         $this->idBroker = new FakeIdBroker($idBrokerUsers);
     }
+    
+    /**
+     * @Then an exception should NOT have been thrown
+     */
+    public function anExceptionShouldNotHaveBeenThrown()
+    {
+        $possibleException = $this->exceptionThrown ?? new Exception();
```

Comment:
My problem was that lines 410 and 411 are evaluated before I know whether `$this->exceptionThrown` is actually an Exception, thus my use of `$possibleException` for (and only for) assembling the error message.

What I'm actually checking for is whether `$this->exceptionThrown` is an exception, in which case I fail the test and show the error message. So no, if any type of exception is thrown (in a context in which I catch it and save it to `$this->exceptionThrown`), this will fail the test as desired.